flepimop/gempyor_pkg/src/gempyor/shared_cli.py

```diff
@@ -0,0 +1,288 @@
+    ),
+    "in_run_id": click.Option(
+        ["--in-id", "in_run_id"],
+        envvar="FLEPI_RUN_INDEX",
+        type=AN_STR,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    "out_run_id": click.Option(
+        ["--out-id", "out_run_id"],
+        envvar="FLEPI_RUN_INDEX",
+        type=AN_STR,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    "in_prefix": click.Option(
+        ["--in-prefix"],
+        envvar="FLEPI_PREFIX",
+        type=AN_STR,
+        default=None,
+        show_default=True,
+        help="unique identifier for the run",
+    ),
+    "first_sim_index": click.Option(
+        ["-i", "--first_sim_index"],
+        envvar="FIRST_SIM_INDEX",
+        type=click.IntRange(min=1),
+        default=1,
+        show_default=True,
+        help="The index of the first simulation",
+    ),
+    "stoch_traj_flag": click.Option(
+        ["--stochastic/--non-stochastic", "stoch_traj_flag"],
+        envvar="FLEPI_STOCHASTIC_RUN",
+        default=False,
+        help="Run stochastic simulations?",
+    ),
+    "write_csv": click.Option(
+        ["--write-csv/--no-write-csv"],
+        default=False,
+        show_default=True,
+        help="write csv output?",
+    ),
+    "write_parquet": click.Option(
+        ["--write-parquet/--no-write-parquet"],
+        default=True,
+        show_default=True,
+        help="write parquet output?",
+    ),
+}
+
+
+def click_helpstring(
+    params: click.Parameter | list[click.Parameter],
+) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
+    """
+    Dynamically append `click.Parameter`s to the docstring of a decorated function.
+
+    Args:
+        params: A parameter or a list of parameters whose corresponding argument
+            values and help strings will be appended to the docstring of the
+            decorated function.
+
+    Returns:
+        The original function with an updated docstring.
+
+    Notes:
+        Adapted from https://stackoverflow.com/a/78533451.
+
+    Examples:
+        >>> import click
+        >>> opt = click.Option("-n", "--number", type=int, help="Your favorite number.")
+        >>> @click_helpstring(opt)
+        ... def test_cli(num: int) -> None:
+        ...     print(f"Your favorite number is {num}!")
+        ...
+        >>> print(test_cli.__doc__)
+
+
+        Command Line Interface arguments:
+
+        n: Int
+    """
+    if not isinstance(params, list):
+        params = [params]
+
+    def decorator(func):
+        # Generate the additional docstring with args from the specified functions
+        additional_doc = "\n\tCommand Line Interface arguments:\n"
+        for param in params:
+            paraminfo = param.to_info_dict()
+            additional_doc += (
+                f"\n\t\t{paraminfo['name']}: {paraminfo['type']['param_type']}\n"
+            )
+
+        if func.__doc__ is None:
+            func.__doc__ = ""
```

Comment:
Might also want to do `additional_doc.lstrip()` to avoid an odd new line + tab here.

Reply:
sure. I think this is going to want some polish, but that feels like its own issue. longer term, we might not even use this, as the CLI side and core functionality should be more modularized imo.